modules/sdk-coin-dot/src/lib/iface.ts

```diff
@@ -142,12 +158,87 @@ export interface AddAnonymousProxyArgs {
  */
 export type BatchCallObject = {
   callIndex: string;
-  args: Record<string, any>;
+  args:
+    | Record<string, any>
+    | AddProxyBatchCallArgs
+    | AddAnonymousProxyBatchCallArgs
+    | StakeBatchCallArgs
+    | StakeMoreCallArgs;
 };
 export interface BatchArgs {
   calls: BatchCallObject[];
 }
 
+export interface AddAnonymousProxyBatchCallArgs {
+  proxy_type: ProxyType;
+  index: number;
+  delay: number;
+}
+
+export interface AddProxyBatchCallArgs {
+  delegate: string | AccountId;
+  proxy_type: ProxyType;
+  delay: number;
+}
+
+export type AccountId = { id: string };
+
+export function isAccountId(value: string | AccountId): value is AccountId {
```

Comment:
Let's move functions out of type definition files.

Reply:
Fixed.